### PR TITLE
ci(lint): setup nonamedreturns linter

### LINKED ISSUE #5455: Add linter option to disallow named returns
### Description

As it's very easy to introduce hard to spot bugs like [this one](https://github.com/kumahq/kuma/pull/5454/files#diff-7d84ed4bad5e95cb5701c0cc1a93d17627a682ef44fc4503425a9b8f95b1d94bR120) I believe we should disallow all named returns in our code base

### PR DESCRIPTION
**Pull request by mmorel-35**
Fixes #5455

### Checklist prior to review

- [x] Link to docs PR or issue --
- [x] Link to UI issue or PR --
- [x] Is the [issue worked on linked][1]? --
- [x] The PR does not hardcode values that might break projects that depend on kuma (e.g. "kumahq" as a image registry) --
- [x] The PR will work for both Linux and Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Unit Tests --
- [x] E2E Tests --
- [x] Manual Universal Tests --
- [x] Manual Kubernetes Tests --
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [x] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

**Comment by lahabana** [comment]
@mmorel-35 thx! I guess you are fixing the failures introduce next?

**Comment by mmorel-35** [comment]
Hi @lahabana , 
Yes, I did just the golangci config for the moment, I'll do the fixes soon

**Comment by mmorel-35** [comment]
HI @lahabana ,
That was a lot of files, I hope I haven't introduced any bug in the way

**Comment by slonka** [comment]
> That is a lot of files... Would there be any way to do it progressively ?

@mmorel-35 - this [issue](https://github.com/golang/go/issues/53215#issue-1259173896) indicates that `singlechecker` (which nonamedreturns is based on) "take package patterns string". So I guess we could do this on a package basis?

**Comment by lahabana** [comment]
@mmorel-35 what do you think about @slonka 's suggestion?

**Comment by lahabana** [comment]
@lobkovilya and @lukidzi please prioritize this PR as it touches a lot of things and it's annoying to keep up to date :)

File: tools/xds-client/main.go [hunk 1]
```diff
@@ -103,7 +103,8 @@ func newRunCmd() *cobra.Command {
 					<-time.After(delay)
 					// proceed
 
-					errCh <- func() (errs error) {
+					errCh <- func() error {
+						var errs error
```
Comment (michaelbeaumont) [inline tools/xds-client/main.go:107]: I don't think this is correct anymore, `errs` is used in deferred functions here but now it's never set whereas before with the named return, it's set to the returned error.

Comment (michaelbeaumont) [reply tools/xds-client/main.go:107]: Such `defers` may be present at other places in this PR, I'll take another look tomorrow.

Comment (michaelbeaumont) [reply tools/xds-client/main.go:107]: Added a commit to change these back. Not quite sure how/whether it's worth getting rid of them for `defer`.

**Review by bartsmykla — APPROVED** [review]

**Pull request merged**